packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-node-ciemss.vue

```diff
@@ -61,12 +76,70 @@ const runResult = ref<DataArray>([]);
 const runResultPre = ref<DataArray>([]);
 const runResultSummary = ref<DataArray>([]);
 const runResultSummaryPre = ref<DataArray>([]);
+const drilldownLossPlot = ref<HTMLElement>();
 
 const csvAsset = shallowRef<CsvAsset | undefined>(undefined);
 
 const areInputsFilled = computed(() => props.node.inputs[0].value && props.node.inputs[1].value);
 const inProgressCalibrationId = computed(() => props.node.state.inProgressCalibrationId);
 
+const selectedOutputId = ref<string>();
+const selectedRunId = computed(() => props.node.outputs.find((o) => o.id === selectedOutputId.value)?.value?.[0]);
+
+let lossValues: { [key: string]: number }[] = [];
+
+function drawLossGraph() {
+	if (drilldownLossPlot.value) {
+		renderLossGraph(drilldownLossPlot.value, lossValues, {
+			width: 200,
+			height: 120
+		});
+	}
+}
+
+async function updateLossChartWithSimulation() {
+	if (props.node.active) {
+		selectedOutputId.value = props.node.active;
+		const simulationObj = await getSimulation(selectedRunId.value);
+		if (simulationObj?.updates) {
+			lossValues = simulationObj?.updates.map((d, i) => ({
+				iter: i,
+				loss: d.data.loss
+			}));
+			drawLossGraph();
+		}
+	}
+}
+
+const messageHandler = (event: ClientEvent<any>) => {
+	lossValues.push({ iter: lossValues.length, loss: event.data.loss });
+	drawLossGraph();
+};
+
+watch(
+	() => props.node.state.inProgressCalibrationId,
+	(id) => {
+		if (_.isEmpty(id)) {
+			unsubscribeToUpdateMessages([id], ClientEventType.SimulationPyciemss, messageHandler);
+		} else {
+			subscribeToUpdateMessages([id], ClientEventType.SimulationPyciemss, messageHandler);
+		}
+	},
+	{ immediate: true }
+);
+
+watch(
+	() => props.node.state.inProgressPreForecastId,
+	() => updateLossChartWithSimulation()
+);
+
+watch(
+	() => props.node.state.inProgressForecastId,
+	() => updateLossChartWithSimulation()
+);
```

Comment:
```suggestion
watch(
	() => [props.node.state.inProgressPreForecastId, props.node.state.inProgressForecastId],
	() => updateLossChartWithSimulation()
);
```

Reply:
This section of watchers are redundant, chatted with Adrian and we clean a few things
- Remove the watchers
- Remove the pub/sub part because it is incomplete (you lose all the events before you subscribe).
- Hook the chart drawing logic to the polling loop, since we have the full loss values under "updates[].data"